attic/multibody/generic_lcm_log_parser.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <string>
+#include "drake/multibody/rigid_body.h"
+#include "attic/multibody/generic_lcm_log_parser.h"
+#include "drake/lcm/drake_lcm_log.h"
+#include "drake/systems/analysis/simulator.h"
+#include "drake/systems/framework/diagram_builder.h"
+#include "drake/systems/lcm/lcm_log_playback_system.h"
+#include "drake/systems/lcm/lcm_subscriber_system.h"
+
+#include "systems/primitives/vector_aggregator.h"
+
+namespace dairlib {
+namespace multibody {
+
+  template<typename T, typename U>
+  void parseLcmLog(const RigidBodyTree<double>& tree,
+      std::string file, std::string channel,
+      Eigen::VectorXd* t, Eigen::MatrixXd* x, double duration = 1.0e6) {
+
+    using drake::systems::lcm::LcmSubscriberSystem;
+
+    drake::lcm::DrakeLcmLog r_log(file, false);
+
+    drake::systems::DiagramBuilder<double> builder;
+
+    builder.AddSystem<drake::systems::lcm::LcmLogPlaybackSystem>(&r_log);
+
+    auto output_sub = builder.AddSystem(
+        LcmSubscriberSystem::Make<T>(channel, &r_log));
+
+    auto output_receiver = builder.AddSystem<U>(tree);
```

Comment:
This requires that `U(tree)` be a constructor. Is it possible to be more general? For instance, what if, the argument was `parseLcmLog(std::unique_ptr<U> system`, then you should be able to call `builder.AddSystem(system)` using this API

https://github.com/RobotLocomotion/drake/blob/master/systems/framework/diagram_builder.h#L56

This would also allow this to apply to any system with vector outputs (and should therefore be moved to `systems/`

Reply:
Do you think the implementation can be extended to `builder.AddSystem(Args &&... args)` as well?

Reply:
It probably could be, but would that be a useful API?

Reply:
Should these files be in `attic/systems/` or `systems/`?